### PR TITLE
Proto api integration

### PR DESCRIPTION
**Pull request by Aaron7317**
Created ControllableState and RobotManager classes. ControllableState stores a signal layout and holds values for those signals. RobotManager stores a Dictionary of ControllableStates and has a ConcurrentQueue for updating them. I changed UpdateMessage to UpdateSignals and changed it to match the format of the mirabuf Signals. I also cleaned up the TcpServer a bit. It now doesn't need locks and its still thread safe (I think). It now has a SetTargetQueue method which sets the queue it will send the updates it recieves to. I made a couple of tests as well but commented out one of them because it uses the TcpServer.

### CODE REVIEW

File: api/Api/Proto/Assembly.cs [hunk 1]
```diff
@@ -0,0 +1,3280 @@
+// <auto-generated>
```
Comment (KyroVibe) [inline api/Api/Proto/Assembly.cs:1]: Add all the generated protobuf files to the gitignore

File: api/Api/Utilities/ControllableState.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+using System.Collections.Generic;
+using System.Text;
+using Mirabuf.Signal;
+using Mirabuf;
+
+namespace SynthesisAPI.Utilities
+{
+    public class ControllableState
+    {
+        private Signals? _currentSignalLayout;
```
Comment (KyroVibe) [inline api/Api/Utilities/ControllableState.cs:12]: If you could add some documentation to the controllable state that would be great. Just like a sentence or two of what it's supposed to do at the top

**Review by KyroVibe — CHANGES_REQUESTED** [review]

**Comment by Aaron7317** [comment]
Made requested changes and changed the Update method in ControllableState to make sure each Signal being updated is part of the current layout

**Review by NickBurrell — APPROVED** [review]

**Review by KyroVibe — APPROVED** [review]

**Comment by NickBurrell** [comment]
Forgot to add a comment to my previous message, but everything looks good. The unit tests are still broken, for somewhat unknown reasons, but this is an issue that we can investigate at a later time (as we planned to).

**Review by matthew17754 — CHANGES_REQUESTED** [review]
For some reason CI checks are stalling and there's still protobuf generated C# files present.

File: api/Api/Proto/Joint.cs [hunk 3]
```diff
@@ -0,0 +1,2132 @@
+// <auto-generated>
+//     Generated by the protocol buffer compiler.  DO NOT EDIT!
+//     source: joint.proto
+// </auto-generated>
```
Comment (matthew17754) [inline api/Api/Proto/Joint.cs:4]: `Joint.cs` should be removed

**Review by matthew17754 — CHANGES_REQUESTED** [review]

**Review by KyroVibe — APPROVED** [review]

**Comment by KyroVibe** [comment]
I added an init script so instead of manually setting up the project, init will initialize the mirabuf submodule and generate all the proto files. Matt if you want you could try doing that for your CI

**Review by matthew17754 — APPROVED** [review]

**Pull request merged**